fe/fe-core/src/main/java/org/apache/doris/catalog/JdbcResource.java

```diff
@@ -144,22 +156,46 @@ public void checkProperties(Map<String, String> properties) throws AnalysisExcep
     protected void setProperties(Map<String, String> properties) throws DdlException {
         Preconditions.checkState(properties != null);
         for (String key : properties.keySet()) {
-            if (!DRIVER_URL.equals(key) && !JDBC_URL.equals(key) && !USER.equals(key) && !PASSWORD.equals(key)
-                    && !TYPE.equals(key) && !DRIVER_CLASS.equals(key)) {
-                throw new DdlException("JDBC resource Property of " + key + " is unknown");
+            switch (key) {
+                case DRIVER_URL:
+                case JDBC_URL:
+                case USER:
+                case PASSWORD:
+                case TYPE:
+                case DRIVER_CLASS:
+                case ONLY_SPECIFIED_DATABASE: // optional argument
+                case LOWER_CASE_TABLE_NAMES: // optional argument
+                    break;
+                default:
+                    throw new DdlException("JDBC resource Property of " + key + " is unknown");
             }
         }
         configs = properties;
+        handleOptionalArguments();
         checkProperties(DRIVER_URL);
         checkProperties(DRIVER_CLASS);
         checkProperties(JDBC_URL);
         checkProperties(USER);
         checkProperties(PASSWORD);
         checkProperties(TYPE);
+        checkProperties(ONLY_SPECIFIED_DATABASE);
+        checkProperties(LOWER_CASE_TABLE_NAMES);
         this.configs.put(JDBC_URL, handleJdbcUrl(getProperty(JDBC_URL)));
         configs.put(CHECK_SUM, computeObjectChecksum(getProperty(DRIVER_URL)));
     }
 
+    /**
+     * This function used to handle optional arguments
+     * eg: only_specified_database、lower_case_table_names
+     */
+    private void handleOptionalArguments() {
+        for (String s : OPTIONAL_PROPERTIES) {
```

Comment:
Not all optional properties using `false` as default.
So better use `if else` or `switch` to handle them separately

Reply:
done